### PR TITLE
Upgrade canton to 0.23.0

### PR DESCRIPTION
**Pull request by oliverse-da**
Updating canton and trying again to move canton dependency from WORKSPACE to deps.bzl

### Pull Request Checklist

- [X] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [X] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: deps.bzl [hunk 1]
```diff
@@ -384,3 +384,19 @@ genrule(
...
+    if "canton" not in native.existing_rules():
+        http_archive(
+            name = "canton",
+            build_file_content = """
+        package(default_visibility = ["//visibility:public"])
+
+        java_import(
+            name = "lib",
+            jars = glob(["lib/**/*.jar"]),
+        )
```
Comment (cocreature) [inline deps.bzl:397]:
```suggestion
package(default_visibility = ["//visibility:public"])

java_import(
    name = "lib",
    jars = glob(["lib/**/*.jar"]),
)
```

**Review by cocreature — APPROVED** [review]
Thanks!

File: deps.bzl (hunk 1 above)
Comment (cocreature) [reply deps.bzl:397]: with that it wroks locally for me so hopefully it also fixes CI

Comment (oliverse-da) [reply deps.bzl]: Thanks - let's give it a try.

Comment (cocreature) [reply deps.bzl]: seems to work now, you’re missing an (empty) changelog entry

Comment (oliverse-da) [reply deps.bzl]: of course

**Pull request merged**